spec/lib/services/migration/failure_manager_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe Migration::FailureManager, in_memory_rails_cache: true do
+  let(:instance) { described_class.new(data_migration:) }
+
+  describe "#record_failures" do
+    subject { instance.record_failure(item, failure_message) }
+
+    context "with correct params" do
+      let(:data_migration) { create(:data_migration, model: :statement, failure_count: 2) }
+      let(:item) { create(:ecf_migration_statement) }
+      let(:failure_message) { "Test failure" }
+
+      it "records a failure" do
+        subject
+
+        failure = Rails.cache.read("migration_failure_#{data_migration.model}_#{data_migration.id}_1")
```

Comment:
Could call `all_failures` here to get it back so we're not relying on an implementation detail